content/en/blog/_posts/2020-01-21-csi-ephemeral-inline-volumes.md

```diff
@@ -0,0 +1,245 @@
+
+```
+The test cluster is ready. Log in with /work/pmem-csi/_work/pmem-govm/ssh-pmem-govm, run kubectl once logged in.
+Alternatively, KUBECONFIG=/work/pmem-csi/_work/pmem-govm/kube.config can also be used directly.
+
+To try out the pmem-csi driver persistent volumes:
+...
+
+To try out the pmem-csi driver ephemeral volumes:
+   cat deploy/kubernetes-1.17/pmem-app-ephemeral.yaml | /work/pmem-csi/_work/pmem-govm/ssh-pmem-govm kubectl create -f -
+```
+
+`deploy/kubernetes-1.17/pmem-app-ephemeral.yaml` specifies one volume:
+
+```
+kind: Pod
+apiVersion: v1
+metadata:
+  name: my-csi-app-inline-volume
+spec:
+  containers:
+    - name: my-frontend
+      image: busybox
+      command: [ "sleep", "100000" ]
+      volumeMounts:
+      - mountPath: "/data"
+        name: my-csi-volume
+  volumes:
+  - name: my-csi-volume
+    csi:
+      driver: pmem-csi.intel.com
+      fsType: "xfs"
+      volumeAttributes:
+        size: "2Gi"
+        nsmode: "fsdax"
+```
+
+Once we have created that pod, we can inspect the result:
+
+```sh
+kubectl describe pods/my-csi-app-inline-volume
+```
+
+```
+Name:         my-csi-app-inline-volume
+...
+Volumes:
+  my-csi-volume:
+    Type:              CSI (a Container Storage Interface (CSI) volume source)
+    Driver:            pmem-csi.intel.com
+    FSType:            xfs
+    ReadOnly:          false
+    VolumeAttributes:      nsmode=fsdax
+                           size=2Gi
+```
+
+```sh
+kubectl exec my-csi-app-inline-volume -- df -h /data
+```
+
+```
+Filesystem                Size      Used Available Use% Mounted on
+/dev/ndbus0region0fsdax/d7eb073f2ab1937b88531fce28e19aa385e93696
+                          1.9G     34.2M      1.8G   2% /data
+```
+
+
+## [Image Populator](https://github.com/kubernetes-csi/csi-driver-image-populator)
+
+The image populator automatically unpacks a container image and makes
+its content available as an ephemeral volume. It's still in
+development, but canary images are already available which can be
+installed with:
+
+```sh
+kubectl create -f https://github.com/kubernetes-csi/csi-driver-image-populator/raw/master/deploy/kubernetes-1.16/csi-image-csidriverinfo.yaml
+kubectl create -f https://github.com/kubernetes-csi/csi-driver-image-populator/raw/master/deploy/kubernetes-1.16/csi-image-daemonset.yaml
+```
+
+This example pod will run nginx and have it serve data that
+comes from the `kfox1111/misc:test` image:
+
+```sh
+kubectl create -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+  - name: nginx
+    image: nginx:1.13-alpine
+    ports:
+    - containerPort: 80
+    volumeMounts:
+    - name: data
+      mountPath: /usr/share/nginx/html
+  volumes:
+  - name: data
+    csi:
+      driver: image.csi.k8s.io
+      volumeAttributes:
+          image: kfox1111/misc:test
+EOF
+```
+
+```sh
+kubectl exec nginx -- cat /usr/share/nginx/html/test
+```
+
+That `test` file just contains a single word:
+```
+testing
+```
+
```

Comment:
Maybe one more little bit:

Data containers can be built with Dockerfiles such as:
```yaml
FROM scratch
COPY index.html /index.html
```

Reply:
Added.